### ISSUE TITLE
Not able to publish bot after upgrading composer to 1.3.0

### ISSUE THREAD

**Issue by NanheGupta** [microsoft/BotFramework-Composer#5456]
Not able to publish bot to Azure using Bot Framework Composer (v1.3.0).
## Describe the bug

When I click [Publish selected bots] it asks to login to Microsoft account.
When I enter my Azure credential it gives the below error.
"This username may be incorrect. Make sure you typed it correctly. Otherwise, contact your admin."

However, I am able to login to Azure portal or Microsoft account using the same credential.
## Version

v1.3.0

## OS
Windows 10 Pro

**Comment by gabog** [comment]
Hi @tonyanziano, can you assist here?

Thanks

**Comment by tonyanziano** [comment]
Hi @NanheGupta ,

If you are trying to login with a personal Microsoft account (non-work / school) then authentication will not work.

This is a configuration issue on our end and we are currently working to fix it, but we have to wait on an external team to make a change.

**Comment by MoesLab** [comment]
Yes, it is my personal, I am guessing I should wait, is there any work around?

**Comment by tonyanziano** [comment]
I believe there is a way to publish manually using the `az` cli, but other than that there currently is no alternative within the Composer UI for publishing with personal accounts.

**Comment by anniexudan** [comment]
I have the same issue. Any solution here?

**Comment by MoesLab** [comment]
Hi Tony, is there any documentation for deploying the chatbot created by the composer from az?
Is there any news from the external team looking into this, I assume it should not be a difficult thing since all work has been done for none-personal accounts? can you enlight us what is the issue the external team having?

**Comment by tonyanziano** [comment]
@MoesLab let me see if I can find some documentation on how to do a manual publish, or if I can get someone on the team more familiar to explain how to do it.

The current situation with the external team is that we are blocked until they make a change on their side, and to make the change, we have to get approval which is taking longer than expected. We are trying to push the approval through as fast as possible.

**Comment by tonyanziano** [comment]
I've checked with the team and there currently is no documentation explaining how to manually do the publish. It is not trivial and requires moving folders and assets around in a certain way.

Sorry for the inconvenience, but we will keep you updated on the approval situation for enabling personal account authentication.

**Comment by srinaath** [comment]
@NanheGupta  This is still on our radar and we will work towards adding the docs for this in our upcoming release. A very rough of set of steps would be

*   move files so that runtime is at root, dialogs are in ComposerDialogs/ folder
*   write secrets to settings file if necessary
*   run lubuild process from cli
*   zip deploy

**Comment by irwinwilliams** [comment]
TL;DR:
Hi, a workaround to get publishing working would be to create a user in the Azure subscription's AD, grant it the appropriate permissions and publish with the login from that user.
Details:
Literally, for the sub you're going to publish into, go into the Active Directory 
![image](https://user-images.githubusercontent.com/1428703/107137566-1b03d400-68e4-11eb-8dd2-628fb32eb5c8.png)
Click Users > + New User
![image](https://user-images.githubusercontent.com/1428703/107137579-3bcc2980-68e4-11eb-824a-90384089ae30.png)
Go back to the Directory, the Users. Find the new user.
Click Assigned Roles
![image](https://user-images.githubusercontent.com/1428703/107137613-7a61e400-68e4-11eb-9871-a5b13eecf0b3.png)

In my case, I gave the user Application administrator and Application developer. 

I was then able to use this user to login to Composer for publishing. You should disable this user/reduce permissions when you're not actively involved in publishing.

**Comment by MoesLab** [comment]
I have created a new user and I used it in the e-mail dialog box to publish the bot.

in Configure resources page the Loading keeps circling when I select Create new Azure resources.
I am thinking I should select Import existing Azure resources. but I don't exactly know what are resources I should have in Azure and what exactly I must add to the JSON  config file. I am hoping somebody can kindly show me an example of it.

**Comment by zxyanliu** [comment]
I happen to see this issue when updating the existing [publish a bot](https://docs.microsoft.com/en-us/composer/how-to-publish-bot) article. The content is still WIP in [this PR](https://review.docs.microsoft.com/en-us/composer/how-to-publish-bot?branch=pr-en-us-375#import-existing-azure-resources).

Pasting the content here for your reference. 

## Import existing Azure resources

This section covers steps to import existing Azure resources. If you don't have your Azure resources provisioned, follow the instructions from the [Create new Azure resources](#create-new-azure-resources) section.

**1**. After your [login to Azure from Composer](#login-to-azure), select **Import existing Azure resources** from the **Configure resources** window.

**2**. Edit the JSON file with the following 2 optional settings in the publish profile:

   1. `hostname` - this is the hostname of your azure webapp or azure function, if it does not match the form of `<name>-<env>`;

   2. `luisResource` - this is the hostname of your LUIS endpoint resource if not in the form `<name>-<env>-<luis>`. You can EXCLUDE cosmos db, applicationInsights, blob storage, or luis if you don't want those features enabled or use them. This is the primary way you can opt-in or opt-out of those features in the runtime.

    - For example, you should use this when deploying the bot without configuring any other services:
  
        ```JSON
        {
        "accessToken": "<your access token>",
        "hostname": "<your web app name>",  
        "settings": {
            "MicrosoftAppId": "<the appid of your bot channel registration>",
            "MicrosoftAppPassword": "<the app password of your bot channel registration>"
            }
        }
        ```

    - If you have LUIS configured, you should use this:

        ```JSON
          {
          "accessToken": "<your access token>",
          "hostname": "<your web app name>",
          "luisResource": "<your luis service name>", 
          "settings": {
              "luis": {
              "endpointKey": "<your luis endpointKey>",
              "authoringKey": "<your luis authoringKey>",
              "region": "<your luis region, for example westus>"
                  },
              "MicrosoftAppId": "<the appid of your bot channel registration>",
              "MicrosoftAppPassword": "<the app password of your bot channel registration>"
              }
          }
        ```

- **If you do not have an access token** or your access token is expired, follow these steps to get a new token:

  - Open a terminal window.
  - Run `az account get-access-token`.
  - This will result in a JSON object containing the new `accessToken`, printed to the console.
  - Copy the value of the accessToken from the terminal and into the publish `accessToken` field in the profile in Composer.


**3**. Paste the edited JSON file from the previous step to the **Publish Configuration** window and select **Save**.

![image](https://user-images.githubusercontent.com/32497439/107713154-797fd800-6c7f-11eb-9cdb-d61b709afeac.png)

**4**. Then follow [the instructions here](https://docs.microsoft.com/en-us/composer/how-to-publish-bot#publish) to publish your bot.

**Comment by tonyanziano** [comment]
This issue might be resolved once #5706 is fixed. I will update this issue when auth for MSA accounts is enabled.
